main.rb

```diff
@@ -0,0 +1,180 @@
+# rubocop: disable Metrics/ModuleLength, Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/MethodLength
+module Enumerable
+  def my_each(&parm)
+    return to_enum :my_each unless block_given?
+
+    item = self
+    i = 0
+    while i < item.length
+      if item.is_a?(Array)
+        yield(item[i])
+      elsif parm.arity == 1
+        yield(assoc item.keys[i])
+      else
+        yield item.keys[i], item[item.keys[i]]
+      end
+      i += 1
+    end
+    item
+  end
+
+  def my_each_with_index(&parm)
+    return to_enum :my_each unless block_given?
+
+    item = self
+    i = 0
+    while i < item.length
+      if item.is_a?(Array)
+        yield item[i], i
+      elsif parm.arity == 1
+        yield(assoc item.keys[i])
+      else
+        yield item.keys[i], item[item.keys[i]], i
+      end
+      i += 1
+    end
+    item
+  end
+
+  def my_select
+    return to_enum :my_select unless block_given?
+
+    items = self
+
+    items = items.to_a if items.is_a? Range
+
+    if items.is_a?(Array)
+
+      new_values = []
+
+      items.my_each { |i| new_values.push(i) if yield(i) == true }
+
+      new_values
+
+    elsif items.is_a?(Hash)
+
+      new_values = {}
+
+      items.my_each { |a, b| new_values.store(a, b) if yield(a, b) == true }
+      new_values
+    else
+      items
+    end
+  end
+
+  def my_all?(parms = nil)
+    item = self
+
+    if parms.nil? && !block_given?
+      item.my_each do |value|
+        results = value.nil? || !value == true
+        break if results
+      end
+      !results
+    elsif parms.nil?
+      item.my_each do |value|
+        results = yield(value)
+        break unless results
+      end
+      results
+    elsif parms.is_a?(Regexp)
+      item.my_each do |value|
+        results = value.to_s.match?(parms)
+        break unless results
+      end
+      results
+    elsif parms.is_a?(Class)
+      item.my_each do |value|
+        results = value.is_a?(parms)
+        break unless results
+      end
+      results
+    else
+      item.my_each do |value|
+        results = value == parms
+        break unless results
+      end
+      results
+    end
+  end
```

Comment:
**`my_all?`** identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2)) 
- Returns true if the block never returns false or nil
```ruby
[1, 2, 3].my_all? { |num| num < 4}   #should return true
```
- when no block or argument is given, returns true when none of the collection members are false or nil
```ruby
[1, 2, 3].my_all? #should return true
[1, 2, nil].my_all? #should return false
[1, false, nil].my_all? #should return false
```
- when a class is passed as an argument returns true if all of the collection is a member of such class
```ruby
[1, 2, 3].my_all?(Integer) #should return true
[1, 2, nil].my_all?(Integer) #should return false
```
- when a Regex is passed as an argument returns true if all of the collection matches the Regex
```ruby
%w[dog door rod blade].my_all?(/d/)   #should return true.
```
- when a pattern other than Regex or a Class is given, returns true if all of the collection matches the pattern
```ruby
[3, 3, 3].my_all?(3) #should return true
[3, 3, 1].my_all?(3) #should return false
```